### ISSUE TITLE
Peak selection plot not showing after loading session

### ISSUE THREAD

**Issue by raimund-schluessler** [BrillouinMicroscopy/BMicro#79]
The problem is, that `circle_fits_interpolation` in the extraction_model is not recreated after opening a file.

**Comment by matthias-baer** [comment]
Ah, right. I will implement some kind of post-creation hook.


Am 05.05.2021 um 11:43 schrieb Raimund Schlüßler ***@***.***>:

﻿

The problem is, that circle_fits_interpolation in the extraction_model is not recreated after opening a file.

**Comment by raimund-schluessler** [comment]
Great. Maybe it's enough to call a refresh/recreate function on every model after the data was loaded.

Other than that the serialization seems to work well. Evaluating a file and saving the data works fine.

When this post-creation hook is available we should also look into moving the calibration interpolation out of the function call and do it only once when the underlying data changes:
https://github.com/BrillouinMicroscopy/bmlab/blob/main/bmlab/models/calibration_model.py#L183-L197
https://github.com/BrillouinMicroscopy/bmlab/blob/main/bmlab/models/calibration_model.py#L151-L161

**Comment by matthias-baer** [comment]
The Serializer class now has a method `post_deserialization` which is empty in the base class and is called after deserialization. When child classes need to do stuff after deserialization, they can override it. I did that in the ExtractionModel to refresh the interpolations. 

It should work, but maybe there is some even nicer way to do that with decorators like in Java EE.

**Comment by matthias-baer** [comment]
Actual issue is fixed with PR #21.

Open: moving interpolation stuff to controller.

**Comment by raimund-schluessler** [comment]
> 
> 
> Actual issue is fixed with PR #21.
> 
> Open: moving interpolation stuff to controller.

Great, thanks a lot.

I created a separate issue for the controller stuff https://github.com/BrillouinMicroscopy/bmlab/issues/22
